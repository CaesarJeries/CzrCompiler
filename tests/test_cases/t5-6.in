void main() {

    //while test

    bool a = true;
    bool e = true;
    int c = 0;
    int d = 0;

    while(a and e)
    {
        d = d + 1;
        c = c + 2;
        a = c > d;
        e = c < d * 2;
    }
    printi(c);
    print("\n");
    printi(d);
    print("\n");


    int ione = 1;
    int itwo = 10;
    bool bone = false;
    bool btwo = false;
    while(ione < itwo or bone or btwo)
    {
        print("true\n");
        printi(ione);
        print("\n");
        printi(itwo);
        print("\n");
        ione = ione + 1;
        bone = ione < 15;
        btwo = (ione + 5) < 17;
    }

    if (btwo) print("true"); else print("false");

}