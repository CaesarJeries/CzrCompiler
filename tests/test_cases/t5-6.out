Loaded: ./exceptions.s
2
1
true
1
10
true
2
10
true
3
10
true
4
10
true
5
10
true
6
10
true
7
10
true
8
10
true
9
10
true
10
10
true
11
10
true
12
10
true
13
10
true
14
10
false